src/renderer/dx/DxRenderer.cpp

```diff
@@ -1047,14 +1047,10 @@ HANDLE DxEngine::GetSwapChainHandle()
 
 void DxEngine::_InvalidateRectangle(const til::rectangle& rc)
 {
-    auto invalidate = rc;
-
-    if (_invalidateFullRows)
-    {
-        invalidate = til::rectangle{ til::point{ static_cast<ptrdiff_t>(0), rc.top() }, til::size{ _invalidMap.size().width(), rc.height() } };
-    }
-
-    _invalidMap.set(invalidate);
+    const auto size = _invalidMap.size();
+    const auto topLeft = til::point{ 0, std::min(size.height(), rc.top()) };
+    const auto bottomRight = til::point{ size.width(), std::min(size.height(), rc.bottom()) };

```

Comment:
I'm sure @miniksa will agree:

i am surprised we do not have `til::rectangle::clamp`

Reply:
I mean we do technically have an `operator&`, but I found it cumbersome to first construct a rectangle (which is rather verbose) just to do `& rc`.

Reply:
Only somewhat verbose :smile:

```
const auto size = _invalidMap.size();
_invalidMap.set(til::rectangle{ 0, rc.top(), size.width(), rc.bottom() } & til::rectangle{ size });
```

okay I admit, it does look somewhat cumbersome.

Reply:
I personally prefer Dustin's, but I'll take either. The structures and operators are tools to make life easier... not a strict requirement.

Reply:
If I'm reading this correctly, there isn't a strong opinion... as I'd like to keep it the way I did it for now, if you don't mind. 😅